pulsar-client/src/main/java/org/apache/pulsar/client/api/Producer.java

```diff
@@ -74,6 +74,27 @@
      */
     CompletableFuture<MessageId> sendAsync(T message);
 
+    /**
+     * Flush all the messages buffered in the client when {@link ProducerBuilder#enableBatching(boolean)} is enabled.
+     * It is a no-op if batching is disabled.
```

Comment:
The behavior could be consistent even when batching is not being used. 

For example, one could always do: 
```java
producer.sendAsync(msg1);
producer.sendAsync(msg2);
producer.sendAsync(msg3);

producer.flush();
```

This should be equivalent to just waiting on the last future.

Reply:
sorry the behavior is consistent. it is waiting on the last future. I will improve the documentation.